### PR TITLE
Bump dependencies

### PR DESCRIPTION
**Pull request by busches**
This bumps almost all the dependencies, except for Stylelint and the two XO configs.

Bumping `github-reserved-names` to 1.1.7 fixes one of the errors called out here: https://github.com/sindresorhus/refined-github/issues/2530 when you visit a sponsor page, e.g. https://github.com/sponsors/sindresorhus

### CODE REVIEW

**Comment by fregante** [comment]
Can you update a few more dependencies if minor so this can be a routine “Update dependencies”?

File: source/libs/utils.ts [hunk 1]
```diff
@@ -138,3 +138,5 @@ export function reportBug(featureName: string, bugName: string): void {
 	console.log('Find existing issues:\n' + String(issuesUrl));
 	console.log('Open new issue:\n' + String(newIssueUrl));
 }
+
+export const noOp = (): void => {}; // eslint-disable-line @typescript-eslint/no-empty-function
```
Comment (sindresorhus) [inline source/libs/utils.ts]:
```suggestion
export const noop = (): void => {}; // eslint-disable-line @typescript-eslint/no-empty-function
```

It's more common to write it like this.

https://api.jquery.com/jQuery.noop/

Comment (fregante) [reply source/libs/utils.ts]: This is the smallest non-optimization ever. Can we just leave it inlined?

**Comment by busches** [comment]
Lol, I didn't do it as an optimization, but to only have to do the eslint
rule suppression once.

**Comment by fregante** [comment]
Yeah I didn’t mean it for you, but in general I haven’t seen a `noop` function declared in so long since arrow functions made it so short. The rule can be disabled in package.json

**Pull request merged**